src/NuGet.Core/NuGet.Commands/RestoreCommand/FrameworkRuntimeDefinition.cs

```diff
@@ -0,0 +1,77 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+#nullable enable
+
+using System;
+using System.Globalization;
+using NuGet.Frameworks;
+using NuGet.Shared;
+
+namespace NuGet.Commands
+{
+    internal sealed record FrameworkRuntimeDefinition : IEquatable<FrameworkRuntimeDefinition>, IComparable<FrameworkRuntimeDefinition>
```

Comment:
Does NuGetFramework have a default comparer?  If so, you don't need to implement IEquatable yourself, the compiler does when you use `record` rather than `class`: https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQskUAJgAQBiUAhgLYwDu0A1gEoCuAdgC4CWDACIwAZny59+ELoXJzyAb1nyVGAMzkMOAAzkAKjSgBzGDwCCAGz40AzovImeAbnIBfQspVz15AHIcAcVNqeiZWKloGZigWe0cXdwJPLx8tXU5eARgASVIYTLEYKDjTBI8CL3lUnT9Qkuc3QkTCVKw/QODIsJjFJqA